hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1677 @@
+    bin_lst, data, flags, nsamples = simple_lst_bin(
+        data=data, 
+        flags=None if ignore_flags else flags,
+        nsamples=nsamples,
+        data_lsts=lsts,
+        baselines=antpairs,
+        lst_bin_edges=lst_bin_edges,
+        freq_array = freqs,
+        rephase = rephase,
+        antpos=antpos,
+    )
+
+    bins = get_lst_bins(lsts, lst_bin_edges)[0]
+    times = np.concatenate(time_arrays)
+    times_in_bins = []
+    for i in range(len(bin_lst)):
+        mask = bins == i
+        times_in_bins.append(times[mask])
+
+    return bin_lst, data, flags, nsamples, times_in_bins
+
+def apply_calfile_rules(
+        data_files: list[list[str]], 
+        calfile_rules: list[tuple[str, str]],
+        ignore_missing: bool
+) -> tuple[list[list[str]], list[list[str]]]:
+    input_cals = []
+    for night, dflist in enumerate(data_files):
+        this = []
+        input_cals.append(this)
+        missing = []
+        for df in dflist:
+            cf = df
+            for rule in calfile_rules:
+                cf = cf.replace(rule[0], rule[1]) 
+
+            if os.path.exists(cf):
+                this.append(cf)
+            elif ignore_missing:
+                warnings.warn(f"Calibration file {cf} does not exist")
+                missing.append(df)
+            else:
+                raise IOError(f"Calibration file {cf} does not exist")
+        data_files[night] = [df for df in dflist if df not in missing]
+    return data_files, input_cals
+
+def filter_required_files_by_times(
+    lst_range: tuple[float, float], 
+    data_metas: list[list[FastUVH5Meta]],
+    cal_files: list[list[str]] | None = None,
+) -> tuple[list, list, list, list, list]:
+
+    lstmin, lstmax = lst_range
+    lstmin %= (2 * np.pi)
+    lstmax %= (2 * np.pi)
+    if lstmin > lstmax:
+        lstmax += 2 * np.pi
+
+    if not cal_files:
+        cal_files = [[None for d in dm] for dm in data_metas]
+
+    tinds = []
+    all_lsts = []
+    file_list = []
+    time_arrays = []
+    cals = []
+
+    # This loop just gets the number of times that we'll be reading.
+    # Even though the input files should already have been configured to be those
+    # that fall within the output LST range, we still need to read them in to
+    # check exactly which time integrations we require.
+    
+    for night, callist in zip(data_metas, cal_files):
+        for meta, cal in zip(night, callist):
+            tarr = meta.times
+            lsts = meta.lsts % (2*np.pi)
+            lsts[lsts < lstmin] += 2 * np.pi
+            
+            tind = (lsts > lstmin) & (lsts < lstmax)
+
+            if np.any(tind):
+                tinds.append(tind)
+                time_arrays.append(tarr[tind])
+                all_lsts.append(lsts[tind])
+                file_list.append(meta)
+                cals.append(cal)
+    return tinds, time_arrays, all_lsts, file_list, cals
+
+
+def make_lst_grid(
```

Comment:
why not load this from lstbin.py?

Reply:
I was doing that, but ended up wanting to have the new module fully distinct from the old, since at some time we must just replace the old one with this.